### ISSUE TITLE
Auxilliary GPG Key needs to be updated for Red Hat 8

### ISSUE THREAD

**Issue by m-usmanayub** [ComplianceAsCode/content#5751]
#### Description of problem:
Ensure Red Hat GPG Key Installed rule is failing in Red Hat 8 because auxiliary key check uses Red Hat 7 Auxiliary key

#### SCAP Security Guide Version:
1.3.1

#### Operating System Version:
RHEL 8.1

#### Steps to Reproduce:

1. import Red Hat GPG Keys from https://access.redhat.com/security/team/key for Red Hat 8
2. Run the scap scan using oscap xccdf eval --profile ospp /usr/share/xml/scap/ssg/content/ssg-rhel8-ds.xml

#### Actual Results:
Red Hat GPG Key Installed rule is failing

#### Expected Results:
Red Hat GPG Key Installed rule should pass

#### Addition Information/Debugging Steps:
The check that is performed is using Red Hat 8 Auxiliary Key which is causing the rule to fail and needs to be updated for Red Hat 8.

**Comment by ggbecker** [comment]
How are you importing this GPG key? Can you run the following command to check the GPG keys installed?
`rpm -q --queryformat "%{SUMMARY},%{RELEASE},%{VERSION}\n" gpg-pubkey`

it should output something like:
```
gpg(Red Hat, Inc. (release key 2) <security@redhat.com>),4ae0493b,fd431d51
gpg(Red Hat, Inc. (auxiliary key) <security@redhat.com>),5b32db75,d4082792
```

As far as I know the fingerprint is correct (from here: [rhel8/product.yml#L20](https://github.com/ComplianceAsCode/content/blob/master/rhel8/product.yml#L20)) and the check should work if you import keys with the following command:
`rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release`

**Comment by m-usmanayub** [comment]
I am using the same command as you mentioned to import the rpm keys and the output is exactly the same as you mentioned however in the scap report generated the rule is failing and when looking at the remediation script I found the following in the beginning

# The two fingerprints below are retrieved from https://access.redhat.com/security/team/key
readonly REDHAT_RELEASE_2_FINGERPRINT="567E347AD0044ADE55BA8A5F199E2F91FD431D51"
readonly REDHAT_AUXILIARY_FINGERPRINT="43A6E49C4A38F4BE9ABF2A5345689C882FA658E0"

The 2nd fingerprint corresponds to Red Hat 7 Auxiliary key probably due to which the rule is failing in the report.

**Comment by ggbecker** [comment]
Ok, can you tell me the version of `scap-security-guide` you are using? Because this was an issue in the past but it was supposed to be fixed.

`rpm -qa scap-security-guide`

This was supposed to be fixed in `0.1.46`. If it is up to date, then try generating the remediation again and check the fingerprint:
`oscap xccdf generate fix --profile ospp /usr/share/xml/scap/ssg/content/ssg-rhel8-ds.xml> remediation`

**Comment by m-usmanayub** [comment]
ah I have the version `scap-security-guide-0.1.42-11.el8.noarch`. Thanks for letting me know I'll update to the latest version and re-run the evaluation. Closing this now.
